rhemyn/compile.js

```diff
@@ -224,8 +249,18 @@ const generateSet = (node, negated, get, stringSize) => {
   }
 
   return [
-    ...out,
-    ...checkFailure()
+    // if len - counter <= 0, break
+	[ Opcodes.local_get, Length ],
+	[ Opcodes.local_get, Counter ],
+	[ Opcodes.i32_sub ],
+	...number(0, Valtype.i32),
+	[ Opcodes.i32_gt_s ],
+	...number(1, Valtype.i32),
+	[ Opcodes.i32_xor ],
+	[ Opcodes.br_if, 0 ], 
```

Comment:
Since this is repeated a lot could we move this to a util function and do like `...checkBounds()`/etc each time instead of copying?